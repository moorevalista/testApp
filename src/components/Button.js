import React from "react"
import { StyleSheet, Text, TouchableOpacity } from "react-native"

export default function Button(props) {

  return (
    <TouchableOpacity style={styles.ButtonPrimaryDefault} onPress={props.setDataMaster} testID='btn'>
      <Text style={styles.Txt745}>Make Master</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  ButtonPrimaryDefault: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    alignSelf: 'center',
    position: 'absolute',
    bottom: '2%',
    paddingVertical: '3%',
    paddingHorizontal: '5%',
    borderRadius: 8,
    backgroundColor: "rgba(54,163,136,1)",
    borderWidth: 1,
    borderStyle: "solid",
    borderColor: "rgba(252,252,252,0.5)",
    backdropFilter: "blur(12px)",
    shadowColor: "rgba(160,163,189,0.16)",
    elevation: 2,
    shadowOffset: { width: 0, height: 16 },
    marginHorizontal: '15%'
  },
  Txt745: {
    flex: 1,
    fontSize: 18,
    fontFamily: "CircularStd-Medium",
    fontWeight: "500",
    lineHeight: 24,
    color: "rgba(252,252,252,1)",
    textAlign: "center",
    justifyContent: "center"
  },

})
