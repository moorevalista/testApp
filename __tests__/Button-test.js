import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../src/components/Button';
import Adapter from 'enzyme-adapter-react-16';
import {shallow, configure} from 'enzyme';

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

configure({adapter: new Adapter(), disableLifecycleMethods: true})

function setDataMaster() {
  console.log('OK');
}

const appWrapper = shallow(<Button setDataMaster={() => setDataMaster()} />)

describe('Button Component', () => {
  
  it('should renders correctly', () => {
    expect(renderer.create(<Button setDataMaster={() => setDataMaster()} />).toJSON()).toMatchSnapshot();
  })

  it('should renders `Button Component` module correctly', () => {
    expect(appWrapper).toMatchSnapshot()
  })

  it('should click when `Make Master` button Pressed', () => {
    const mockCallBack = jest.fn()
    appWrapper.find('[testID="btn"]').simulate('press')
    expect(mockCallBack.mock.calls.length)
  })
})