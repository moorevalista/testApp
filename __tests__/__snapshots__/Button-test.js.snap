// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should renders \`Button Component\` module correctly 1`] = `ShallowWrapper {}`;

exports[`Button Component should renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backdropFilter": "blur(12px)",
      "backgroundColor": "rgba(54,163,136,1)",
      "borderColor": "rgba(252,252,252,0.5)",
      "borderRadius": 8,
      "borderStyle": "solid",
      "borderWidth": 1,
      "bottom": "2%",
      "display": "flex",
      "elevation": 2,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginHorizontal": "15%",
      "opacity": 1,
      "paddingHorizontal": "5%",
      "paddingVertical": "3%",
      "position": "absolute",
      "shadowColor": "rgba(160,163,189,0.16)",
      "shadowOffset": Object {
        "height": 16,
        "width": 0,
      },
    }
  }
  testID="btn"
>
  <Text
    style={
      Object {
        "color": "rgba(252,252,252,1)",
        "flex": 1,
        "fontFamily": "CircularStd-Medium",
        "fontSize": 18,
        "fontWeight": "500",
        "justifyContent": "center",
        "lineHeight": 24,
        "textAlign": "center",
      }
    }
  >
    Make Master
  </Text>
</View>
`;
